java/arcs/core/storage/DriverFactory.kt

```diff
@@ -57,24 +54,19 @@ object DriverFactory {
      * instances. Therefore using this method requires shutting down all arcs, and should be use
      * only in rare circumstances.
      */
-    suspend fun removeAllEntities(): Job = coroutineScope {
-        launch {
-            providers.value.forEach { it.removeAllEntities() }
-        }
+    suspend fun removeAllEntities() {
+        providers.value.forEach { it.removeAllEntities() }
     }
 
     /**
      * Clears all entities created in the given time range. See comments on [removeAllEntities] re
      * the need to recreate stores after calling this method.
      */
-    suspend fun removeEntitiesCreatedBetween(startTimeMillis: Long, endTimeMillis: Long): Job =
-        coroutineScope {
```

Comment:
with the current code, if one of the provider fails, it would not cancel the parent scope. With you change, I think one provider failing would cancel the parent scope, ie the other providers as well? is that true/what we want?

Reply:
it would still cancel the parent scope, unless that parent scope was a SupervisorScope.

Reply:
just for my understanding, do you have a reference for this (canceling of the parent)?
According to https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/coroutine-scope.html " When any child coroutine in this scope fails, this scope fails and all the rest of the children are cancelled". So it looks me other children (as well as this) are cancelled, but the parent is not.

Anyway I agree that here we should use supervisorScope. Thanks!

Reply:
Yeah... I was puzzled for a bit by this, I think the documentation is a bit misleading here. It took my a few moments to come up with a response here that I felt happy with, and I'm curious to hear @jasonwyatt's thoughts as well.

 It's weird that they specifically call out that the child jobs of the `coroutineScope`. I guess it's to indicate that `coroutineScope` doesn't change the job cancellation behavior for its child jobs. But they don't specifically mention that the enclosing scope will also behave as it normally would.

So if a child job of `coroutineScope` fails, then the newly created scope fails... but this also means its parent will fail, etc (until a SupervisorScope is hit). 

The value in `coroutineScope` mainly seems to be to allow you to easily start a number of jobs in parallel and await their completion, but it doesn't change anything about cancellation behavior in the job tree.

(I had to verify with a few tests in Kotlin playground just to make sure).

Reply:
I see, thanks for the explanation!